apps/examples/araplayer/HLSParser.cpp

```diff
@@ -0,0 +1,281 @@
+#include "HLSParser.h"
+#include <algorithm>
+#include <vector>
+#include <map>
+
+#include "AudioUtils.h"
+
+const char *HLSParser::PLAYLIST_EXT = ".m3u8";
+const char *HLSParser::SEGMENT_EXT = ".ts";
+const char *HLSParser::TARGET_DURATION = "#EXT-X-TARGETDURATION";
+const char *HLSParser::BEGIN_PLAYLIST_URL = "#EXT-X-STREAM-INF";
+const char *HLSParser::BEGIN_SEGMENT_URL = "#EXTINF";
+const char *HLSParser::BANDWIDTH = "BANDWIDTH";
+
+HLSParser::hls_play_list_t *HLSParser::findPlaylist(const char *url)
+{
+	for (std::map<std::string, hls_play_list_t *>::iterator iter = mMapMasterPlaylist.begin(); iter != mMapMasterPlaylist.end(); ++iter) {
+		if (iter->first.compare(url) == 0) {
+			return iter->second;
+		}
+
+		std::map<std::string, hls_play_list_t *> mapMediaPlaylist = iter->second->mapMediaPlaylist;
+		for (std::map<std::string, hls_play_list_t *>::iterator iterMedia = mapMediaPlaylist.begin(); iterMedia != mapMediaPlaylist.end(); ++iterMedia) {
+			if (iterMedia->first.compare(url) == 0) {
+				return iterMedia->second;
+			}
+		}
+	}
+	return NULL;
+}
+
+bool HLSParser::isFirstParsedMediaPlaylist(hls_play_list_t *mediaPlaylist)
```

Comment:
validation check or get number of parsed? what is correct? you want to check playlist which has checked only MASTER, right?

Reply:
Playlist is a multi-tree structure. determined whether it's a master or not(single). It's also used to determine if the sub playlist is parsed.

Reply:
Ok but you are using index as a boolean type, I mean if you want to keep index number, then below style is needed.
```
if (idx > 0 && idx < 3) {
   do_something_A;
} else if (idx > 3) {
   do_something_B:
}
```
So, if you want to use index as a boolean, you can stop finding, when you found it.